Makefile

```diff
@@ -0,0 +1,46 @@
+app_name=notifications
+
+project_dir=$(CURDIR)/../$(app_name)
+build_dir=$(CURDIR)/build
+source_dir=$(build_dir)/$(app_name)
+sign_dir=$(build_dir)/sign
+
+all: package
+
+dev-setup: clean npm-update build-js
+
+npm-update:
+	rm -rf node_modules
+	npm update
+
+build-js:
+	npm run dev
+
+build-js-production:
+	npm run build
+
+clean:
+	rm -rf $(build_dir)
+
+package: clean build-js-production
+	mkdir -p $(source_dir)
+	rsync -a \
+	--exclude=/build \
+	--exclude=/docs \
+	--exclude=/js-src \
+	--exclude=/l10n/.tx \
+	--exclude=/tests \
+	--exclude=/.git \
+	--exclude=/.github \
+	--exclude=/CONTRIBUTING.md \
+	--exclude=/issue_template.md \
+	--exclude=/README.md \
+	--exclude=/.gitignore \
+	--exclude=/.scrutinizer.yml \
+	--exclude=/.travis.yml \
+	--exclude=/.drone.yml \
+	--exclude=/node_modules \
+	--exclude=/npm-debug.log \
+	--exclude=/package.json \
```

Comment:
if you're using an up-to-date version of npm, it will generate a `package-lock.json`. You might want to exclude that too.

Reply:
dont have that on npm 5.6.0

Reply:
Okay, maybe it's just generated if you use `npm install --save xxx` or similar https://docs.npmjs.com/files/package-lock.json

Reply:
Hmm let me add this, so everyone is installing the same stuff?

Reply:
Exactly, it locks down the exact version number, just like composer does with `composer.lock`

Reply:
Oh, and it speeds up `npm install` as npm doesn't have to build the dependency graph 👍